webrtc/RTCTransport-helper.js

```diff
@@ -0,0 +1,184 @@
+// to turn to connected or completed state. When the RTCPeerConnectionState
+// is "connected", all RTCIceTransports and RTCDtlsTransports are in the
+// "connected", "completed" or "closed" state and at least one of them is in the
+// "connected" or "completed" state.
+function waitConnectingPc(pc) {
+  return new Promise((resolve, reject) => {
+    if (pc.connectionState === 'connected') {
+      resolve();
+    }
+
+    pc.addEventListener('connectionstatechange', () => {
+      const { connectionState } = pc;
+
+      if (connectionState === 'connected') {
+        resolve();
+      } else if (['closed', 'failed'].includes(connectionState)) {
+        reject(new Error(`one of DTLS/ICE transports transition to unexpected state ${connectionState}`));
+      }
+    });
+  });
+}
+
+function getDtlsTransportFromSctpTransport(sctpTransport) {
+  assert_true(sctpTransport instanceof RTCSctpTransport,
+    'Expect pc.sctp to be instantiated from RTCSctpTransport');
+
+  const dtlsTransport = sctpTransport.transport;
+  assert_true(dtlsTransport instanceof RTCDtlsTransport,
+    'Expect sctp.transport to be an RTCDtlsTransport');
+
+  return dtlsTransport;
+}
+
+function getIceTransportFromSctpTransport(sctpTransport) {
+  const dtlsTransport = getDtlsTransportFromSctpTransport(sctpTransport);
+
+  const iceTransport = dtlsTransport.transport;
+  assert_true(iceTransport instanceof RTCIceTransport,
+    'Expect dtlsTransport.transport to be an RTCIceTransport');
+
+  return iceTransport;
+}
+
+// Get one or more RTCDtlsTransports from RTCRtpSender/Receiver
+// in pc. There may be more than one e.g. there are more than one
+// senders/receivers or when there are different underlying
+// transports for RTP and RTCP.
+function getDtlsTransportsFromSenderReceiver(pc) {
+  const senders = pc.getSenders();
+  const receivers = pc.getReceivers();
+
+  const dtlsTransportsSet = new Set();
+
+  function adddDtlsTransport(transport) {
+    // Add a dtls transport to the result set, if it is
+    // not null/undefined. Although the spec mandates both
+    // transport and rtcpTransport fields must be set together,
+    // we validate that requirement in separate test cases.
+    if (transport) {
+      assert_true(transport instanceof RTCDtlsTransport,
+        'Expect transport to be an RTCDtlsTransport');
+
+      dtlsTransportsSet.add(transport);
+    }
+  }
+
+  for (const sender of senders) {
+    adddDtlsTransport(sender.transport);
+    adddDtlsTransport(sender.rtcpTransport);
+  }
+
+  for (const receiver of receivers) {
+    adddDtlsTransport(receiver.transport);
+    adddDtlsTransport(receiver.rtcpTransport);
+  }
+
+  const dtlsTransports = [...dtlsTransportsSet];
+
+  if (dtlsTransports.length === 0) {
+    assert_unreached('Expect to get at least one unique RTCDtlsTransport from sender receivers');
+  }
+
+  return dtlsTransports;
+}
+
+function getIceTransportFromDtlsTransport(dtlsTransport) {
```

Comment:
This function is so small, I don't think it should be a function.

You can replace it with dtlsTransport.transport (dtlsTransport.iceTransport after the change agreed to in Lyon goes through).